shell/components/CopyToClipboardText.vue

```diff
@@ -12,18 +12,38 @@ export default {
     }
   },
 
+  data() {
+    return { copied: false };
+  },
+
   methods: {
     clicked(event) {
-      event.preventDefault();
-      this.$copyText(this.text);
+      if (!this.copied) {
+        event.preventDefault();
+        this.$copyText(this.text);
+        this.copied = true;
+
+        let t = event.target;
+
+        if (t.tagName === 'I') {
+          t = t.parentElement || t;
+        }
+        t.classList.add('copied');
+        setTimeout(() => {
+          try {
+            t.classList.remove('copied');
+            this.copied = false;
+          } catch (_e) {}
+        }, 500);
+      }
     },
   }
 };
 </script>
 
 <template>
   <a class="copy-to-clipboard-text" :class="{'plain': plain}" href="#" @click="clicked">
-    {{ text }} <i class="icon icon-copy" />
+    {{ text }} <i v-if="!copied" class="icon icon-copy" /><i v-else class="icon icon-checkmark" />
```

Comment:
This could be a little neater and easier to test in e2e with 
```
{{ text }} <i class="icon" :class="{ 'icon-copy': !copied, 'icon-checkmark': copied}" />
```